### PR TITLE
docs,READMEs: updates for composability changes

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Updated docs to reflect #5351 and composability changes. Not to be merged before #5351 is released.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
gonna need a rebase here too

**Review by ryanvazquez — APPROVED** [review]

**Comment by Rugvip** [comment]
Ready for review and merge now, release is out

**Review by OrkoHunter — APPROVED** [review]
🙏

File: docs/overview/architecture-overview.md [hunk 1]
```diff
@@ -67,33 +67,41 @@ is available at
...
 ```
 
+Note that we use `"/catalog"` as our path to this plugin page, but we can choose
+any route we want for the page, as long as it doesn't collide with the routes
+that we choose for the other plugins in the app.
+
+These components that are exported from plugins are referred to as "Plugin
+Extension Components", or "Extensions Components". They are regular React
+components, but in addition to being able to be rendered by React, they also
+contain various pieces of metadata that is used to write together the entire
```
Comment (freben) [inline docs/overview/architecture-overview.md:100]:
```suggestion
contain various pieces of metadata that is used to wire together the entire
```

File: docs/overview/architecture-overview.md [hunk 2]
```diff
@@ -67,33 +67,41 @@ is available at
...
+app. Extensions components are created using `create*Extension` methods, which
```
Comment (freben) [inline docs/overview/architecture-overview.md:101]:
```suggestion
app. Extension components are created using `create*Extension` methods, which
```

File: docs/overview/architecture-overview.md [hunk 3]
```diff
@@ -67,33 +67,41 @@ is available at
...
+  </FlatRoutes>
+);
 ```
 
+Note that we use `"/catalog"` as our path to this plugin page, but we can choose
+any route we want for the page, as long as it doesn't collide with the routes
+that we choose for the other plugins in the app.
+
+These components that are exported from plugins are referred to as "Plugin
+Extension Components", or "Extensions Components". They are regular React
```
Comment (freben) [inline docs/overview/architecture-overview.md:98]:
```suggestion
Extension Components", or "Extension Components". They are regular React
```

File: docs/features/techdocs/getting-started.md [hunk 4]
```diff
@@ -29,14 +29,8 @@ yarn add @backstage/plugin-techdocs
...
 
-Add the following to `packages/app/src/plugins.ts`:
-
-```typescript
-export { plugin as TechDocs } from '@backstage/plugin-techdocs';
-```
-
-Now we can add a route for the TechDocs page. In `packages/app/src/App.tsx`,
-import TechDocsPage and add the following to `FlatRoutes`:
+In `packages/app/src/App.tsx`, import TechDocsPage and add the following to
```
Comment (freben) [inline docs/features/techdocs/getting-started.md:32]:
```suggestion
In `packages/app/src/App.tsx`, import `TechDocsPage` and add the following to
```

File: plugins/api-docs/README.md [hunk 5]
```diff
@@ -33,17 +33,7 @@ To link that a component provides or consumes an API, see the [`providesApis`](h
...
-```ts
-// packages/app/src/plugins.ts
-
-export { apiDocsPlugin } from '@backstage/plugin-api-docs';
-```
-
-<Route path="/api-docs" element={<ApiExplorerPage />} />
-
-3. Register the `ApiExplorerPage` at the `/api-docs` path:
+2. Add the `ApiExplorerPage` extensions to the app:
```
Comment (freben) [inline plugins/api-docs/README.md:36]:
```suggestion
2. Add the `ApiExplorerPage` extension to the app:
```

File: plugins/catalog-import/README.md [hunk 6]
```diff
@@ -23,15 +23,7 @@ Some features are not yet available for all supported Git providers.
...
-2. Add the plugin to the app:
-
-```ts
-// packages/app/src/plugins.ts
-
-export { catalogImportPlugin } from '@backstage/plugin-catalog-import';
-```
-
-3. Register the `CatalogImportPage` at the `/catalog-import` path:
+2. Add the `CatalogImportPage` extensions to the app:
```
Comment (freben) [inline plugins/catalog-import/README.md:26]:
```suggestion
2. Add the `CatalogImportPage` extension to the app:
```

File: plugins/rollbar/README.md [hunk 7]
```diff
@@ -12,33 +12,25 @@ Website: [https://rollbar.com/](https://rollbar.com/)
...
+    <EntityLayout.Route path="/rollbar" title="Rollbar">
+      <EntityRollbarContent />
+    </EntityLayout.Route>
+    ...
   </EntityPageLayout>
 );
 ```
 
-5. Setup the `app.config.yaml` and account token environment variable
+4. Setup the `app.config.yaml` and account token environment variable
```
Comment (freben) [inline plugins/rollbar/README.md:33]:
```suggestion
4. Setup the `app-config.yaml` and account token environment variable
```

and same below

**Review by freben — APPROVED** [review]

File: plugins/bitrise/README.md [hunk 8]
```diff
@@ -11,12 +11,6 @@ Welcome to the Bitrise plugin!
 $ yarn add @backstage/plugin-bitrise
 ```
 
-Then make sure to export the plugin in your app's [`plugins.ts`](https://github.com/backstage/backstage/blob/master/packages/app/src/plugins.ts) to enable the plugin:
-
-```js
-export { bitrisePlugin } from '@backstage/plugin-bitrise';
-```
-
```
Comment (freben) [inline plugins/bitrise/README.md:19]: one more thing. mine adds some more changes

https://github.com/backstage/backstage/pull/5452/files

is something in there of interest to also address as part of this pr?

Comment (Rugvip) [reply plugins/bitrise/README.md:19]: Cherry-pick'd and resolved conflicts

**Pull request merged**